src/fit.rs

```diff
@@ -0,0 +1,455 @@
+            * c0
+            - c1 * (9. * s1 - 70. * c1 * mx - 5. * c1 * area));
+    let a1 = 16.
+        * (((12. * s0 - 5. * c0 * (42. * mx - 17. * area)) * s1
+            - 70. * c1 * (3. * mx - area) * s0
+            - 75. * c0 * c1 * area * area)
+            * s1
+            - 75. * c1 * c1 * area * area * s0);
+    let a0 = 80. * s1 * (42. * s1 * mx - 25. * area * (s1 - c1 * area));
+    // TODO: "roots" is not a good name for this variable, as it also contains
+    // the real part of complex conjugate pairs.
+    let mut roots = ArrayVec::<f64, 4>::new();
+    const EPS: f64 = 1e-12;
+    if a4.abs() > EPS {
+        let a = a3 / a4;
+        let b = a2 / a4;
+        let c = a1 / a4;
+        let d = a0 / a4;
+        if let Some(quads) = factor_quartic_inner(a, b, c, d, false) {
+            for (qc1, qc0) in quads {
+                let qroots = solve_quadratic(qc0, qc1, 1.0);
+                if qroots.is_empty() {
+                    // Real part of pair of complex roots
+                    roots.push(-0.5 * qc1);
+                } else {
+                    roots.extend(qroots);
+                }
+            }
+        }
+    } else if a3.abs() > EPS {
+        roots.extend(solve_cubic(a0, a1, a2, a3));
+    } else {
+        roots.extend(solve_quadratic(a0, a1, a2));
+    }
+
+    let s01 = s0 * c1 + s1 * c0;
+    roots
+        .iter()
+        .filter_map(|&d0| {
+            let (d0, d1) = if d0 > 0.0 {
+                let d1 = (d0 * s0 - area * (10. / 3.)) / (0.5 * d0 * s01 - s1);
+                if d1 > 0.0 {
+                    (d0, d1)
+                } else {
+                    (s1 / s01, 0.0)
+                }
+            } else {
+                (0.0, s0 / s01)
+            };
+            if d0 >= 0.0 && d1 >= 0.0 {
+                Some(CubicBez::new(
+                    (0.0, 0.0),
+                    (d0 * c0, d0 * s0),
+                    (1.0 - d1 * c1, d1 * s1),
+                    (1.0, 0.0),
+                ))
+            } else {
+                None
+            }
+        })
+        .collect()
+}
+
+const HUGE: f64 = 1e12;
+
+/// Generate a highly optimized Bézier path that fits the source curve.
+///
+/// This function is still experimental and the signature might change; it's possible
+/// it might become a method on the [`ParamCurveFit`] trait.
+///
+/// This function currently only handles continuous curves, and does not handle cusps
+/// or corners. It also hasn't been rigorously tested.
+///
+/// This function is considerably slower than [`fit_to_bezpath`], as it computes
+/// optimal subdivision points. Its result is expected to be very close to the optimum
+/// possible Bézier path for the source curve, in that it has a minimal number of curve
+/// segments, and a minimal error over all paths with that number of segments.
+pub fn fit_to_bezpath_opt(source: &impl ParamCurveFit, accuracy: f64) -> BezPath {
```

Comment:
I think this still fails when the offset path has a cusp (original path curvature = offset). Not sure if you want to fix it before or after landing but I thought I'd mention.

Reply:
Yes, this is expected, and could probably be explained better. The "source curve" here is actually the offset, so when that has a discontinuity it won't work well. It's easy for people to imagine the "source curve" being the original cubic. I'll try to write a bit more clarification before landing this, then hope to do more followup.